packages/ckeditor5-dev-utils/src/git/commit.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * @license Copyright (c) 2003-2025, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import { simpleGit } from 'simple-git';
+
+const CHUNK_LENGTH_LIMIT = 4000;
+
+export default async function commit(
+	{ cwd, message, files, dryRun = false }: { cwd: string; message: string; files: Array<string>; dryRun?: boolean }
+): Promise<void> {
+	if ( !files.length ) {
+		return;
+	}
+
+	const git = simpleGit( { baseDir: cwd } );
+
+	const makeCommit = async () => {
+		for ( const chunk of splitPathsIntoChunks( files ) ) {
+			await git.add( chunk );
+		}
+
+		return git.commit( message );
+	};
+
+	if ( dryRun ) {
+		const lastCommit = await git.log( [ '-1' ] );
+
+		await makeCommit();
```

Comment:
I don't remember why we want to commit the changes in dry run mode. Isn't it enough to just log the to be committed files?